File: 0169-majority-element/0169-majority-element.java
class Solution {
    public int majorityElement(int[] nums) {
        int count = 1;
        int result = nums[0];
        for (int i = 1; i < nums.length; i++) {
            if (count == 0) {
                result = nums[i];
            }
            count += result == nums[i] ? 1 : -1;
        }

        return result;
    }
}

File: 0169-majority-element/0169-majority-element.cpp
class Solution {
public:
    int majorityElement(vector<int>& nums) {
        int resultCount = 1;
        int result = nums[0];
        for (int i = 1; i < nums.size(); i++) {
            if (nums[i] != result) {
                resultCount--;
                if (resultCount < 0) {
                    result = nums[i];
                    resultCount = 1;
                }
            } else {
                resultCount++;
            }
        }

        return result;
    }
};